board/views.py

```diff
@@ -24,18 +28,79 @@ def __init__(self):
         self.platforms = res_platforms
 
 
+class BaseGameView:
+    def __init__(self, user):
+        self.user = user
+
+    def _get_context(self, game_id, is_favourite):
+        if Game.is_exist(game_id):
+            game = Game(game_id)
+            tweets = Tweet.get_tweets(game_id)
+        else:
+            game = GameAPI(game_id)
+            tweet_ids = twitter_wrapper.get_tweets_by_string(game.slug)
+            if tweet_ids:
+                tweets = [TweetAPI(tweet_id) for tweet_id in tweet_ids]
+            else:
+                tweets = None
+
+        context = {
+            'game': game,
+            'tweets': tweets,
+            'user': self.user,
+            'is_favourite': is_favourite
+        }
+        return context
+
+
+class DetailView(BaseGameView):
+    def get_detail(self, game_id):
+        is_favourite = False
+        if self.user.is_authenticated:
+            is_favourite = bool(self.user.favourite_games.filter(game_id=game_id).first())
+        context = self._get_context(game_id, is_favourite)
+        return context
+
+
+class FavouriteView(View):
+    def _data_init(self):
+        data = json.loads(self.request.body.decode())
+        self.game_id = data['game_id']
+        self.favourite_games = self.request.user.favourite_games
+
+    def post(self, request, *args, **kwargs):
+        self._data_init()
+        request.user.favourite_games.update_or_create(game_id=self.game_id, user=request.user)
+        return HttpResponse(status=200)
+
+    def delete(self, request, *args, **kwargs):
+        self._data_init()
+        favourite_game = request.user.favourite_games.filter(game_id=self.game_id).first()
+        if favourite_game:
+            favourite_game.delete()
+        return HttpResponse(status=200)
+
+
 def main(request):
     data = request.GET
+    platforms = [item for item in data.getlist('platforms')]
+    genres = [item for item in data.getlist('genres')]
+    rating = data.get('rating', default=None)
 
-    platforms = [int(item) for item in data.getlist('platforms')]
-    genres = [int(item) for item in data.getlist('genres')]
-    rating = [int(item) for item in data.getlist('rating')]
-    res = igdb_wrapper.get_games(platforms=platforms, genres=genres, rating=rating)
-
-    games = []
-    if res:
-        for game in res:
-            games.append(Game(game['id']))
+    if rating:
+        rating = int(rating)
+    games = Game.get_games(platforms=platforms, genres=genres, rating=rating)
+    if games:
+        print('Games are here, in DB!!!')
+    if not games:
+        print('There are no games in DB!')
+        print(platforms, genres, rating)
+        res = igdb_wrapper.get_games(platforms=platforms, genres=genres, rating=[rating,])
```

Comment:
Let's just return the games available from database, without calling IGDB API in this view anymore.